src/common/login/AppLock.ts

```diff
@@ -27,4 +30,14 @@ export class MobileAppLock implements AppLock {
 		}
 		return this.mobileSystemFacade.enforceAppLock(await this.mobileSystemFacade.getAppLockMethod())
 	}
+
+	/**
+	 * If the selected AppLockMethod is not present in supportedAppLockMethods allow fallback to `None`, otherwise
+	 * it will keep on restricting user from loggingIn
+	 */
+	async setAppLockMethod() {
```

Comment:
at this point we have cleared the credentials, I think this method should be called `resetAppLockMethod()` and it should unconditionally set to `None`